pkg/kubelet/cadvisor/cadvisor_linux.go

```diff
@@ -117,6 +124,33 @@ func New(port uint, runtime string) (Interface, error) {
 	return cadvisorClient, nil
 }
 
+func generateCollectorHttpClient(cAdvisorCollectorClientCertFile, cAdvisorCollectorClientPrivateKeyFile string) (*http.Client, error) {
+	//Enable accessing insecure endpoints. We should be able to access metrics from any endpoint
+	tlsConfig := &tls.Config{
+		InsecureSkipVerify: true,
```

Comment:
I'm having trouble parsing out whether you're pulling or pushing, but if someone has chosen to set up a secured transport, not verifying that transport defies expectation.  It is worse that using http, because it is deceptive.

If the problem is one of having to use IPs and being unable to get certs for them, I could probably buy that, but I'd expect the claim to be related by number 2, not number 1. 

Unverify https is really bad and in some ways worse that http, because the lack of verification is unexpected.